src/ui/public/filter_bar/react/filter_bar.tsx

```diff
@@ -0,0 +1,216 @@
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { EuiButtonEmpty, EuiFlexGroup, EuiFlexItem, EuiPopover } from '@elastic/eui';
+import { FormattedMessage, InjectedIntl, injectI18n } from '@kbn/i18n/react';
+import classNames from 'classnames';
+import React, { Component } from 'react';
+import chrome from 'ui/chrome';
+import { IndexPattern } from 'ui/index_patterns';
+import { FilterOptions } from 'ui/search_bar/components/filter_options';
+import {
+  disableFilter,
+  enableFilter,
+  MetaFilter,
+  pinFilter,
+  toggleFilterDisabled,
+  toggleFilterNegated,
+  unpinFilter,
+} from '../filters';
+import { buildEmptyFilter } from '../filters/meta_filter';
+import { FilterEditor } from './filter_editor';
+import { FilterItem } from './filter_item';
+
+const config = chrome.getUiSettingsClient();
+
+interface Props {
+  filters: MetaFilter[];
+  onFiltersUpdated: (filters: MetaFilter[]) => void;
+  className: string;
+  indexPatterns: IndexPattern[];
+  intl: InjectedIntl;
+}
+
+interface State {
+  isAddFilterPopoverOpen: boolean;
+}
+
+class FilterBarUI extends Component<Props, State> {
+  public state = {
+    isAddFilterPopoverOpen: false,
+  };
+
+  public render() {
+    const classes = classNames('globalFilterBar', this.props.className);
+
+    return (
+      <EuiFlexGroup
+        className="globalFilterGroup"
+        gutterSize="none"
+        alignItems="flexStart"
+        responsive={false}
+      >
+        <EuiFlexItem className="globalFilterGroup__branch" grow={false}>
+          <FilterOptions
+            onEnableAll={this.onEnableAll}
+            onDisableAll={this.onDisableAll}
+            onPinAll={this.onPinAll}
+            onUnpinAll={this.onUnpinAll}
+            onToggleAllNegated={this.onToggleAllNegated}
+            onToggleAllDisabled={this.onToggleAllDisabled}
+            onRemoveAll={this.onRemoveAll}
+          />
+        </EuiFlexItem>
+
+        <EuiFlexItem>
+          <EuiFlexGroup
+            className={classes}
+            wrap={true}
+            responsive={false}
+            gutterSize="xs"
+            alignItems="center"
+          >
+            {/* TODO display pinned filters first*/}
+            {this.renderItems()}
+            {this.renderAddFilter()}
+          </EuiFlexGroup>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }
+
+  private renderItems() {
+    return this.props.filters.map((filter, i) => (
+      <EuiFlexItem key={i} grow={false}>
```

Comment:
FWIW, assigning the index value to the key _might_ [cause errors when React updates the DOM](https://reactjs.org/docs/lists-and-keys.html#keys). Is there a way to use the filter value itself or some other unique value as the key instead?